core/execution.go

```diff
@@ -293,17 +312,18 @@ func Create(env vm.Environment, caller vm.ContractRef, code []byte, gas, gasPric
 			contract.UseGas(contract.Gas)
 		}
 	}
-
-	// When there are no errors but the maxCodeSize is still exceeded, makes more sense than just failing
+	// Assign err if contract code size exceeds the max while the err is still empty.
 	if maxCodeSizeExceeded && err == nil {
 		err = errMaxCodeSizeExceeded
 	}
 
 	//if there's an error we return nothing
 	if err != nil && err != vm.ErrRevert {
-		return nil, address, err
+		return nil, *address, err
```

Comment:
Why is address used as a pointer in this function by the way? Was there a reason this was changed?